### ISSUE TITLE
cluster/{gce,gke}/util.sh test-setup creates network, but test-teardown doesn't delete it

### ISSUE THREAD

**Issue by ixdy** [kubernetes/kubernetes#21426]
Merging #21122 has caused a lot of Jenkins builds to fail once, since they report that we've leaked networks, default routes, and default firewalls. This is because we create the GCP network if it doesn't exist yet, but we never tear it down.

Is this by design? Or should we be tearing down the test network when we tear down the rest of the cluster?

@kubernetes/goog-testing @kubernetes/goog-cluster

**Comment by bprashanth** [comment]
> Is this by design? Or should we be tearing down the test network when we tear down the rest of the cluster?

I think in cases where multiple suites run in a project, others might be using the e2e network. Kube-up runs with the default network unless otherwise specified so I don't think we really *need* to tear down the e2e network evertime, other than because it feels like the right thing to do.

Routes and firewall rules are a different story. The route controller re-creates routes on every kube-up invocation, so we should probably do the same in our test environemtns. We also shouldn't create the blanket node-ports firewall rule in e2e cluster. This is another thing that will probably break existing tests if we remove, though.

**Comment by ixdy** [comment]
To make it clear which resources the script complained about leaking:
```
++ diff -sw -U0 '-F^\[.*\]$' /var/lib/jenkins/jobs/kubernetes-e2e-gce/workspace/_artifacts/gcp-resources-before.txt /var/lib/jenkins/jobs/kubernetes-e2e-gce/workspace/_artifacts/gcp-resources-after.txt
--- /var/lib/jenkins/jobs/kubernetes-e2e-gce/workspace/_artifacts/gcp-resources-before.txt	2016-02-17 13:11:24.901962510 -0800
+++ /var/lib/jenkins/jobs/kubernetes-e2e-gce/workspace/_artifacts/gcp-resources-after.txt	2016-02-17 13:29:39.118741060 -0800
@@ -32,0 +33,2 @@ [ routes ]
+default-route-73c69a574d1d6b3f jenkins-e2e 0.0.0.0/0     default-internet-gateway 1000
+default-route-fc162100e993caf6 jenkins-e2e 10.240.0.0/16                          1000
@@ -40,0 +43,2 @@ [ firewall-rules ]
+jenkins-e2e-default-internal            jenkins-e2e 10.0.0.0/8    tcp:1-65535,udp:1-65535,icmp
+jenkins-e2e-default-ssh                 jenkins-e2e 0.0.0.0/0     tcp:22
++ [[ true == \t\r\u\e ]]
++ echo '!!! FAIL: Google Cloud Platform resources leaked while running tests!'
```

**Comment by ixdy** [comment]
fixed since #34577
